pkg/proxy/config/config.go

```diff
@@ -98,14 +101,14 @@ func (c *EndpointSliceConfig) RegisterEventHandler(handler EndpointSliceHandler)
 
 // Run waits for cache synced and invokes handlers after syncing.
 func (c *EndpointSliceConfig) Run(stopCh <-chan struct{}) {
-	klog.InfoS("Starting endpoint slice config controller")
+	c.logger.Info("Starting endpoint slice config controller")
```

Comment:
Should we be passing a `context` to Run() instead of a `chan`? Although then there's still no way to get the logger to the handlers... so maybe just keep the whole `ctx` in `EndpointSliceConfig`, etc, rather than just the logger, and then use that ctx here?

Reply:
In go docs https://pkg.go.dev/context, it is advised to *not* store the context in a struct ("Do not store Contexts inside a struct type; instead, pass a Context explicitly..."). I am unsure of the implications, so it may be ok here.

Reply:
Storing a context inside a struct leads to ambiguity: what is the "right" context for a method when it is called from someone else? The context in the struct or the context that the caller is using?

It's better to change all functions and methods to accept a context parameter.